### PR TITLE
Duplicate item

### PR DESCRIPTION
**Pull request by Meklo**

### CODE REVIEW

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java [hunk 1]
```diff
@@ -361,6 +361,24 @@ public void testFilterToScript() throws Exception {
...
+        FormFilter lineFormFilter = new FormFilter(
+                filterId1,
+                creationDate,
+                modificationDate,
+                lineFilter
+        );
+        insertFilter(filterId1, lineFormFilter);
+
+        mvc.perform(post("/" + FilterApi.API_VERSION + "/filters?duplicateFrom=" + filterId1 + "&id=" + UUID.randomUUID())).andExpect(status().isOk());
+
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:379]: Add check on duplicated filter with checkFormFilter method ?

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 2]
```diff
@@ -99,6 +99,17 @@ public <F extends AbstractFilter> AbstractFilter createFilter(F filter) {
         return getRepository(filter).insert(filter);
     }
 
+    @Transactional
+    public Optional<AbstractFilter>  createFilter(UUID parentFilterId, UUID filterId) {
+        Optional<AbstractFilter> parentFilterOptional = getFilter(parentFilterId);
+        if (parentFilterOptional.isPresent()) {
+            AbstractFilter parentFilter = parentFilterOptional.get();
+            parentFilter.setId(filterId);
+            return Optional.of(getRepository(parentFilter).insert(parentFilter));
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/filter/server/FilterService.java]: Maybe use here existing method :  createFilter(F filter)

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by AbdelHedhili — APPROVED** [review]

File: src/main/java/org/gridsuite/filter/server/FilterController.java [hunk 3]
```diff
@@ -66,6 +66,16 @@ public ResponseEntity<AbstractFilter> createFilter(@RequestParam("id") UUID filt
             .body(service.createFilter(filter));
     }
 
+    @PostMapping(value = "/filters")
+    @Operation(summary = "Create a filter from an existing filter")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The filter has been successfully created")})
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/filter/server/FilterController.java:71]: add 404 not found

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 4]
```diff
@@ -99,6 +99,17 @@ public <F extends AbstractFilter> AbstractFilter createFilter(F filter) {
         return getRepository(filter).insert(filter);
     }
 
+    @Transactional
+    public Optional<AbstractFilter>  createFilter(UUID sourceFilterId, UUID filterId) {
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:103]: remove 1 white space

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

**Review by AbdelHedhili — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**